pages/chaoss/visualizations/contrib_importance_over_time.py

```diff
@@ -0,0 +1,443 @@
+                                            ],
+                                            value="Commit",
+                                            clearable=False,
+                                        ), 
+                                         dbc.Alert(
+                                            children="""No contributions of this type have been made.\n
+                                            Please select a different contribution type.""",
+                                            id=f"check-alert-{PAGE}-{VIZ_ID}",
+                                            dismissable=True,
+                                            fade=False,
+                                            is_open=False,
+                                            color="warning",
+                                        ),
+                                    ], 
+                                    className="me-2",
+                                    width=3,
+                                ),
+                            ], 
+                            align = "center"
+                        ),  
+                        dbc.Row(
+                            [
+                                dbc.Label(
+                                    "Threshold:",
+                                    html_for=f"threshold-{PAGE}-{VIZ_ID}",
+                                    width="auto",
+                                ),
+                                dbc.Col(
+                                        [dcc.Slider(
+                                            id=f"threshold-{PAGE}-{VIZ_ID}",  
+                                            min=0.10, 
+                                            max=0.90, 
+                                            step=0.10, 
+                                            value = 0.50,   
+                                        ),    
+                                    ], 
+                                    className="me-2",
+                                    width=6,
+                                ),
+                            ],
+                            align="center",
+                        ),
+                        dbc.Row(
+                            [
+                                dbc.Col(
+                                    dcc.DatePickerRange(
+                                        id=f"date-picker-range-{PAGE}-{VIZ_ID}",
+                                        min_date_allowed=dt.date(2005, 1, 1),
+                                        max_date_allowed=dt.date.today(),
+                                        initial_visible_month=dt.date(dt.date.today().year, 1, 1),
+                                        clearable=True,
+                                    ),
+                                    width="auto",)
+                            ],
+                        ), 
+                        dbc.Row(
+                            [   dbc.Label(
```

Comment:
Curious to hear @JamesKunstle thoughts on this as well but I think the "window width and step size" might be over complicating this a bit and might be confusing for users. It took me a while to be able to understand what was happening and I have been involved in this development process

Reply:
Since those are a part of the analysis, I'd prefer to keep them and document them well, and have the defaults be sensible. Having a hard-coded default doesn't make sense, nor does removing this part of the analysis.